api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -128,7 +128,11 @@ private void validateDataSetCreateRequest(DataSetRequest dataSetRequest) {
       throw new BadRequestException("Missing name");
     } else if (dataSetRequest.getConceptSetIds() == null
         || (dataSetRequest.getConceptSetIds().isEmpty()
-            && dataSetRequest.getPrePackagedConceptSet().equals(PrePackagedConceptSetEnum.NONE))) {
+            && (dataSetRequest.getPrePackagedConceptSet().size() == 1
+                && dataSetRequest
+                    .getPrePackagedConceptSet()
+                    .get(0)
+                    .equals(PrePackagedConceptSetEnum.NONE)))) {
       throw new BadRequestException("Missing concept set ids");
     } else if ((dataSetRequest.getCohortIds() == null || dataSetRequest.getCohortIds().isEmpty())
```

Comment:
I would replace this with
`CollectionUtils.isEmpty(dataSetRequest.getConceptSetIds())`